### PR TITLE
Launchcop fixes RJ Robinson

### PR DESCRIPTION
**Pull request by rjrobinson**
Hey guys. Hope this helps. My git log should show the changes. 

Below was the latest offenses. I did run UAT's before hand, and it showed you had 4 errors before i started. I didn't alter that. Looks good. Let me know if there is any thing else I can do 

Offenses:

```
app/controllers/application_controller.rb:51:5: C: Redundant begin block detected.
    begin
    ^^^^^
app/decorators/question_decorator.rb:9:81: C: Line is too long. [90/80]
    'queue-number' + (object.question_queue.status == 'in-progress' ? ' in-progress' : '')
                                                                                ^^^^^^^^^^
app/helpers/question_queue_helper.rb:4:81: C: Line is too long. [152/80]
      link_to "I'm on it!", question_queue_path(id: question_queue, question_queue: { status: "in-progress" }), method: :patch, class: "ee-button on-it"
                                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/helpers/question_queue_helper.rb:6:81: C: Line is too long. [152/80]
      link_to "We Solved It!", question_queue_path(id: question_queue, question_queue: { status: "done" }), method: :patch, class: "ee-button solved-it"
                                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/mailers/comment_mailer.rb:8:81: C: Line is too long. [83/80]
      "There are #{comment_count} new comments on your #{lesson.title} submission."
                                                                                ^^^
app/models/question_queue.rb:7:81: C: Line is too long. [91/80]
    ->(team) { where(team: team).where.not(status: 'done').sort_by { |qq| qq.sort_order } }
                                                                                ^^^^^^^^^^^
bin/spring:14:81: C: Line is too long. [88/80]
    ENV["GEM_PATH"] = ([Bundler.bundle_path.to_s] + Gem.path).join(File::PATH_SEPARATOR)
                                                                                ^^^^^^^^
config/environments/production.rb:76:27: C: Prefer single-quoted strings inside interpolations.
    from: "no-reply@#{ENV["HOST"]}"
                          ^^^^^^
config/initializers/redis.rb:9:81: C: Line is too long. [81/80]
Redis.current = Redis.new(host: uri.host, port: uri.port, password: uri.password)
                                                                                ^
db/migrate/20140928165428_create_lessons.rb:20:81: C: Line is too long. [88/80]
        to_tsvector('english', title || ' ' || body || ' ' || coalesce(description, ''))
                                                                                ^^^^^^^^
db/migrate/20140928165428_create_lessons.rb:31:81: C: Line is too long. [89/80]
      tsvector_update_trigger(searchable, 'pg_catalog.english', title, body, description)
                                                                                ^^^^^^^^^
db/migrate/20140928175407_remove_articles.rb:22:81: C: Line is too long. [125/80]
    ActiveRecord::Base.connection.execute("CREATE INDEX index_challenges_on_searchable ON challenges USING gin (searchable)")
                                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
db/migrate/20140928175407_remove_articles.rb:24:81: C: Line is too long. [234/80]
    ActiveRecord::Base.connection.execute("CREATE TRIGGER challenges_searchable_update BEFORE INSERT OR UPDATE ON challenges FOR EACH ROW EXECUTE PROCEDURE tsvector_update_trigger('searchable', 'pg_catalog.english', 'title', 'body')")
                                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
db/migrate/20140928175407_remove_articles.rb:38:81: C: Line is too long. [230/80]
    ActiveRecord::Base.connection.execute("CREATE TRIGGER articles_searchable_update BEFORE INSERT OR UPDATE ON articles FOR EACH ROW EXECUTE PROCEDURE tsvector_update_trigger('searchable', 'pg_catalog.english', 'title', 'body')")
                                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
db/migrate/20140928175407_remove_articles.rb:40:81: C: Line is too long. [121/80]
    ActiveRecord::Base.connection.execute("CREATE INDEX index_articles_on_searchable ON articles USING gin (searchable)")
                                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/markdown_renderer.rb:12:3: W: Useless private access modifier.
  private
  ^^^^^^^
spec/controllers/answers_controller_spec.rb:9:14: C: Avoid using {...} for multi-line blocks.
      expect {
             ^
spec/controllers/comments_controller_spec.rb:54:14: C: Align the parameters of a method call if they span more than one line.
             format: :json,
             ^^^^^^^^^^^^^^
spec/controllers/questions_controller_spec.rb:69:14: C: Avoid using {...} for multi-line blocks.
      expect {
             ^
spec/features/queue_spec.rb:34:5: W: Useless assignment to variable - question. Did you mean question2?
    question = FactoryGirl.
    ^^^^^^^^
spec/features/queue_spec.rb:42:5: W: Useless assignment to variable - question2. Did you mean question?
    question2 = FactoryGirl.
    ^^^^^^^^^
spec/features/queue_spec.rb:67:5: W: Useless assignment to variable - question.
    question = FactoryGirl.
    ^^^^^^^^
spec/features/view_assignment_status_on_dashboard_spec.rb:35:5: W: Useless assignment to variable - user_comment.
    user_comment = FactoryGirl.create(:comment,
    ^^^^^^^^^^^^
spec/features/view_assignment_status_on_dashboard_spec.rb:47:5: W: Useless assignment to variable - admin_comment.
    admin_comment = FactoryGirl.create(:comment,
    ^^^^^^^^^^^^^
spec/features/view_submissions_spec.rb:37:7: W: Useless assignment to variable - my_submission.
      my_submission = FactoryGirl.
      ^^^^^^^^^^^^^
spec/mailers/comment_mailer_spec.rb:10:81: C: Line is too long. [90/80]
        expected = "There is 1 new comment on your #{submission.lesson.title} submission."
                                                                                ^^^^^^^^^^
spec/mailers/comment_mailer_spec.rb:28:81: C: Line is too long. [85/80]
      let(:comments) { FactoryGirl.create_list(:comment, 3, submission: submission) }
                                                                                ^^^^^
spec/mailers/comment_mailer_spec.rb:32:81: C: Line is too long. [92/80]
        expected = "There are 3 new comments on your #{submission.lesson.title} submission."
                                                                                ^^^^^^^^^^^^
spec/models/question_spec.rb:10:9: W: Useless assignment to variable - q1.
        q1 = FactoryGirl.create(:question, question_queue: qq1)
        ^^
spec/models/question_spec.rb:59:16: C: Avoid using {...} for multi-line blocks.
        expect {
               ^
spec/models/question_spec.rb:70:16: C: Avoid using {...} for multi-line blocks.
        expect {
               ^

214 files inspected, 31 offenses detected
```

### CODE REVIEW

File: app/decorators/question_decorator.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ def queue_id
   end
 
   def queue_progress
-    'queue-number' + (object.question_queue.status == 'in-progress' ? ' in-progress' : '' )
+    'queue-number' + (object.question_queue.status == 'in-progress' ? ' in-progress' : '')
```
Comment (rjrobinson) [inline app/decorators/question_decorator.rb]: I wasn't able to RF this line. I am guessing that it is a ternary op ( spencer? ) . I think a regular if/else block would be fine.

**Comment by dpickett** [comment]
hey RJ - thank you so much! Are you running the latest version of launchcop? I'm still seeing 31 offenses.

```no-highlight
app/controllers/application_controller.rb:51:5: C: Redundant begin block detected.
    begin
    ^^^^^
app/decorators/question_decorator.rb:9:81: C: Line is too long. [90/80]
    'queue-number' + (object.question_queue.status == 'in-progress' ? ' in-progress' : '')
                                                                                ^^^^^^^^^^
app/helpers/question_queue_helper.rb:4:81: C: Line is too long. [152/80]
      link_to "I'm on it!", question_queue_path(id: question_queue, question_queue: { status: "in-progress" }), method: :patch, class: "ee-button on-it"
                                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/helpers/question_queue_helper.rb:6:81: C: Line is too long. [152/80]
      link_to "We Solved It!", question_queue_path(id: question_queue, question_queue: { status: "done" }), method: :patch, class: "ee-button solved-it"
                                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/mailers/comment_mailer.rb:8:81: C: Line is too long. [83/80]
      "There are #{comment_count} new comments on your #{lesson.title} submission."
                                                                                ^^^
app/models/question_queue.rb:7:81: C: Line is too long. [91/80]
    ->(team) { where(team: team).where.not(status: 'done').sort_by { |qq| qq.sort_order } }
                                                                                ^^^^^^^^^^^
bin/spring:14:81: C: Line is too long. [88/80]
    ENV["GEM_PATH"] = ([Bundler.bundle_path.to_s] + Gem.path).join(File::PATH_SEPARATOR)
                                                                                ^^^^^^^^
config/environments/production.rb:76:27: C: Prefer single-quoted strings inside interpolations.
    from: "no-reply@#{ENV["HOST"]}"
                          ^^^^^^
config/initializers/redis.rb:9:81: C: Line is too long. [81/80]
Redis.current = Redis.new(host: uri.host, port: uri.port, password: uri.password)
                                                                                ^
db/migrate/20140928165428_create_lessons.rb:20:81: C: Line is too long. [88/80]
        to_tsvector('english', title || ' ' || body || ' ' || coalesce(description, ''))
                                                                                ^^^^^^^^
db/migrate/20140928165428_create_lessons.rb:31:81: C: Line is too long. [89/80]
      tsvector_update_trigger(searchable, 'pg_catalog.english', title, body, description)
                                                                                ^^^^^^^^^
db/migrate/20140928175407_remove_articles.rb:22:81: C: Line is too long. [125/80]
    ActiveRecord::Base.connection.execute("CREATE INDEX index_challenges_on_searchable ON challenges USING gin (searchable)")
                                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
db/migrate/20140928175407_remove_articles.rb:24:81: C: Line is too long. [234/80]
    ActiveRecord::Base.connection.execute("CREATE TRIGGER challenges_searchable_update BEFORE INSERT OR UPDATE ON challenges FOR EACH ROW EXECUTE PROCEDURE tsvector_update_trigger('searchable', 'pg_catalog.english', 'title', 'body')")
                                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
db/migrate/20140928175407_remove_articles.rb:38:81: C: Line is too long. [230/80]
    ActiveRecord::Base.connection.execute("CREATE TRIGGER articles_searchable_update BEFORE INSERT OR UPDATE ON articles FOR EACH ROW EXECUTE PROCEDURE tsvector_update_trigger('searchable', 'pg_catalog.english', 'title', 'body')")
                                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
db/migrate/20140928175407_remove_articles.rb:40:81: C: Line is too long. [121/80]
    ActiveRecord::Base.connection.execute("CREATE INDEX index_articles_on_searchable ON articles USING gin (searchable)")
                                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/markdown_renderer.rb:12:3: W: Useless private access modifier.
  private
  ^^^^^^^
spec/controllers/answers_controller_spec.rb:9:14: C: Avoid using {...} for multi-line blocks.
      expect {
             ^
spec/controllers/comments_controller_spec.rb:54:14: C: Align the parameters of a method call if they span more than one line.
             format: :json,
             ^^^^^^^^^^^^^^
spec/controllers/questions_controller_spec.rb:69:14: C: Avoid using {...} for multi-line blocks.
      expect {
             ^
spec/features/queue_spec.rb:34:5: W: Useless assignment to variable - question. Did you mean question2?
    question = FactoryGirl.
    ^^^^^^^^
spec/features/queue_spec.rb:42:5: W: Useless assignment to variable - question2. Did you mean question?
    question2 = FactoryGirl.
    ^^^^^^^^^
spec/features/queue_spec.rb:67:5: W: Useless assignment to variable - question.
    question = FactoryGirl.
    ^^^^^^^^
spec/features/view_assignment_status_on_dashboard_spec.rb:35:5: W: Useless assignment to variable - user_comment.
    user_comment = FactoryGirl.create(:comment,
    ^^^^^^^^^^^^
spec/features/view_assignment_status_on_dashboard_spec.rb:47:5: W: Useless assignment to variable - admin_comment.
    admin_comment = FactoryGirl.create(:comment,
    ^^^^^^^^^^^^^
spec/features/view_submissions_spec.rb:37:7: W: Useless assignment to variable - my_submission.
      my_submission = FactoryGirl.
      ^^^^^^^^^^^^^
spec/mailers/comment_mailer_spec.rb:10:81: C: Line is too long. [90/80]
        expected = "There is 1 new comment on your #{submission.lesson.title} submission."
                                                                                ^^^^^^^^^^
spec/mailers/comment_mailer_spec.rb:28:81: C: Line is too long. [85/80]
      let(:comments) { FactoryGirl.create_list(:comment, 3, submission: submission) }
                                                                                ^^^^^
spec/mailers/comment_mailer_spec.rb:32:81: C: Line is too long. [92/80]
        expected = "There are 3 new comments on your #{submission.lesson.title} submission."
                                                                                ^^^^^^^^^^^^
spec/models/question_spec.rb:10:9: W: Useless assignment to variable - q1.
        q1 = FactoryGirl.create(:question, question_queue: qq1)
        ^^
spec/models/question_spec.rb:59:16: C: Avoid using {...} for multi-line blocks.
        expect {
               ^
spec/models/question_spec.rb:70:16: C: Avoid using {...} for multi-line blocks.
        expect {
               ^

214 files inspected, 31 offenses detected
```

**Comment by rjrobinson** [comment]
Hey Dan. I Tried to remove as many as possible. I didn't want to delete extra variables, not knowing the full domain of their purpose. The other "long line" errors required changes to the testing suite as well. 
I do have the latest version.  I can look over it again when i get home.  the "Useless assignment to variable" a listed as warnings. and i am unsure how to deal with the mulit-line blocks as some of them have method calls on them and a do/end block would only hinder this. 

The first one, the ternary operator should be refactored. 

I also didn't want to touch the 6 AR migrations.

**Comment by dpickett** [comment]
ok thanks for the explanations! Really appreciate it man!

**Pull request merged**